src/vivarium/framework/lookup.py

```diff
@@ -257,38 +260,44 @@ class LookupTableInterface:
     def __init__(self, manager):
         self._lookup_table_manager = manager
 
-    def build_table(self, data, key_columns=('sex',), parameter_columns=(['age', 'age_group_start', 'age_group_end'],
-                                                                         ['year', 'year_start', 'year_end']),
-                    value_columns=None) -> LookupTable:
+    def build_table(self, data,
+                    key_columns: Union[List[str], Tuple[str]]=('sex',),
+                    parameter_columns: Union[List[str], Tuple[str]]=(['age', 'age_group_start', 'age_group_end'],
```

Comment:
we have a choice here. This typing is not currently correct. Iterable and Sequence can't distinguish a string from an iter/seq of strings (https://stackoverflow.com/questions/44912374/python-type-annotation-for-sequences-of-strings-but-not-for-strings). To be exact, the typing info would be insane. We could make a type var or leave it inexact in two ways, with iterable or precluding the list of list situation with bin edges (which is always)

Reply:
hmm yeah the parameter columns typing should be list/tuple of lists/tuples of strings. I would tend to not go for iterable and just use lists/tuples and make a typevar so it's not insanely long in the function def. I'm a little confused about what you mean by precluding the list of list situation though?